### PR TITLE
 EN-2343 Split name from id in DE configuration

### PR DESCRIPTION
**Pull request by zonia3000**
I ended using `RandomStringUtils.randomAlphanumeric(ID_LENGTH)` for generating the DE identifier keeping the control on the string length.

Moreover I left also the possibility to specify the DE id manually. If the id it is not found in the payload it is generated automatically. This is very useful for testing the API with Postman or if we want to define a default DE instance. Of course the appbuilder should let the backend generating the id for it.

### CODE REVIEW

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by zonia3000** [comment]
Retest this please

File: plugins/entando-plugin-jpdigital-exchange/src/main/java/org/entando/entando/aps/system/services/digitalexchange/DigitalExchangesServiceImpl.java [hunk 1]
```diff
@@ -102,6 +108,18 @@ public void delete(String digitalExchangeName) {
         return client.getCombinedResult(new TestExchangesCall());
     }
 
+    private void validateName(DigitalExchange digitalExchange) {
+
+        if (getDigitalExchanges().stream()
+                .filter(de -> digitalExchange.getName().equals(de.getName()))
+                .findAny().isPresent()) {
```
Comment (rzsouza) [inline plugins/entando-plugin-jpdigital-exchange/src/main/java/org/entando/entando/aps/system/services/digitalexchange/DigitalExchangesServiceImpl.java]: Can be simplified to:

```
        if (getDigitalExchanges().stream()
                .anyMatch(de -> digitalExchange.getName().equals(de.getName()))) {
```

**Comment by entando-service** [comment]
All is Well

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
All is Well

**Pull request merged**